src/Experiment/tests/InterfaceAssembly.cs

```diff
@@ -0,0 +1,72 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+using System.Reflection.Emit.Experimental;
+using System.Reflection.Metadata;
+using System.Reflection.PortableExecutable;
+using Xunit;
+
+namespace Experiment.Tests
+{
+    public class InterfaceAssembly
+    {
+        const String interfaceName = "EmptyInterface";
+        [Fact]
+        public void GenerateInterfaceAssemblyReadToDisk()
+        {
+            AssemblyName _name = new AssemblyName("MyInterfaceAssembly");
+            _name.Version = new Version("4.3");
+            const String _fileName = "MyInterfaceAssembly.dll";
+            AssemblyBuilder assemblyBuilder = AssemblyBuilder.DefineDynamicAssembly(_name, System.Reflection.Emit.AssemblyBuilderAccess.Run);
+            //We need to create module because even a blank assembly has one module.
+            ModuleBuilder moduleBuilder = assemblyBuilder.DefineDynamicModule(_name.Name);
+            moduleBuilder.DefineType(interfaceName, TypeAttributes.Interface);
+            assemblyBuilder.Save(_fileName);
+            TryLoadAssembly(_fileName,_name);//It seems that sometimes ILSpy will point out a DLL is malformed but MetadataReader doesn't throw an error (MetadataLoadContext rather?) .
+        }  
+    private static void TryLoadAssembly(string filePath, AssemblyName _name)
```

Comment:
This method could be moved to a test utilities class so it can be used on both tests